### PR TITLE
sql: use PG error code for txn errors

### PR DESCRIPTION
**Pull request by rafiss**
Epic: None
Release note: None

### CODE REVIEW

**Review by fqazi — APPROVED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/108203)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ecwall and @michae2)

File: pkg/ccl/logictestccl/tests/local/BUILD.bazel [hunk 1]
```diff
@@ -12,7 +12,7 @@ go_test(
         "//c-deps:libgeos",  # keep
         "//pkg/ccl/logictestccl:testdata",  # keep
     ],
-    shard_count = 19,
```
Comment (mgartner) [inline pkg/ccl/logictestccl/tests/local/BUILD.bazel:15]: Mistake?

**Review by mgartner — APPROVED** [review]
LGTM

File: pkg/ccl/logictestccl/tests/local/BUILD.bazel (hunk 1 above)
Comment (rafiss) [reply pkg/ccl/logictestccl/tests/local/BUILD.bazel:15]: i suppose so. i must have something saved locally that caused this when i did `./dev gen`

**Pull request merged**